builtin/read_osh.py

```diff
@@ -157,9 +157,12 @@ def _ReadPortion(delim_byte, max_chars, cmd_ev):
     ch_array = []  # type: List[int]
     eof = False
 
-    bytes_read = 0
+    chars_read = 0
+    ch_backslash = mylib.ByteAt('\\', 0)
+    ch_newline = mylib.ByteAt('\n', 0)
```

Comment:
Another thing is the recommended way to get the code of a character. `pyos.NEWLINE_CH` seems to be hardcoded to be 10, but do you assume ASCII (for 7-bit characters)?

Reply:
Yes I think we should create `BACKSLASH_CH = 92  # \ ` to be consistent (unfortunately `ord('\\')` will not work as a constant at the top level, but it's OK if we comment it

---

Oils is assuming UTF-8, so `\n` is always 10.  I hope that assumption will not bite us as far as building distros ...

I believe the OpenBSD shell did this - they only support UTF-8

I know this conflicts with libc, which supports multiple encodings (and we still have to fix inconsistency there)

But I think it reduces the number of bugs, and so far no user has complained.  I hope that will continue to be the case!

Reply:
It also sorta relates to my philosophy about a "distributed shell"

I hope the same shell script will mean the same thing on different machines, irrespective of global variables like LANG

But this can also cause incompatibility

We might have to support LANG=C and LANG=utf-8, but that shouldn't affect NEWLINE=10 or BACKSLASH=92